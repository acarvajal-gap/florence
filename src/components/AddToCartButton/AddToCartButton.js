import React from 'react'
import './AddToCartButton.scss'

function AddToCartButton ({ setMattressCount, mattressCount }) {
  return (
    <button className='button is-medium is-fullwidth add-to-cart' onClick={() => setMattressCount(mattressCount + 1)}>
      Add to Cart
    </button>
  )
}

export default AddToCartButton
