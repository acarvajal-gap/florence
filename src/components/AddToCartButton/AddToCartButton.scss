.add-to-cart {
  background-color: $whiskey;
  color: $white;
}