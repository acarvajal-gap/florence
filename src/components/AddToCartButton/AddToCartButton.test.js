import React from 'react'
import { shallow } from 'enzyme'
import AddToCartButton from './AddToCartButton'

const mockProps = {
  setMattressCount: jest.fn(),
  mattressCount: 1
}
const wrapper = shallow(<AddToCartButton {...mockProps} />)

describe('AddToCartButton', () => {
  it('should render the AddToCartButton Component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('call setMattressCount with mattressCount + 1', () => {
    wrapper.find('button').simulate('click')
    expect(mockProps.setMattressCount).toHaveBeenCalledWith(mockProps.mattressCount + 1)
  })
})
