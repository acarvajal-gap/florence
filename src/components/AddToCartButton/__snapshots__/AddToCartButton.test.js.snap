// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton should render the AddToCartButton Component correctly 1`] = `
<button
  className="button is-medium is-fullwidth add-to-cart"
  onClick={[Function]}
>
  Add to Cart
</button>
`;
